### PR TITLE
Adding support for multilingual MonoT5 reranker

### PR DESCRIPTION
**Pull request by lhbonifacio**
Adding support for multilingual MonoT5 reranker.
Using AutoModelForSeq2SeqLM() for loading T5/mT5 models.
A list containing the models name with its predictions tokens was added to help users.

### CODE REVIEW

File: pygaggle/rerank/transformer.py
Comment (rodrigonogueira4) [inline pygaggle/rerank/transformer.py:86]: maybe remove the underscore from here and allow the user to add it when initializing the model? In this code, there is no way of _not_ using underscores in case the model wasn't trained with them.

File: pygaggle/rerank/transformer.py
Comment (rodrigonogueira4) [inline pygaggle/rerank/transformer.py:97]: move this code to the __init__ since these tokens shouldn't change once the model is initialized?

File: pygaggle/rerank/transformer.py
Comment (rodrigonogueira4) [inline pygaggle/rerank/transformer.py:35]: add the underscores here and remove the automatic addition in `get_prediction_tokens`?

**Review by lhbonifacio — COMMENTED** [review]
Requested changes

File: pygaggle/rerank/transformer.py
Comment (rodrigonogueira4) [inline pygaggle/rerank/transformer.py:62]: This needs to be removed, right?

File: pygaggle/rerank/transformer.py
Comment (rodrigonogueira4) [inline pygaggle/rerank/transformer.py:57]: this line seems wrong as `self.token_true` is receiving the value of `token_false`?

File: pygaggle/rerank/transformer.py
Comment (rodrigonogueira4) [inline pygaggle/rerank/transformer.py:86]: maybe call these `token_false_id` so it does not get confused with `token_false`? Same for the __init__ function

File: pygaggle/rerank/transformer.py
Comment (rodrigonogueira4) [inline pygaggle/rerank/transformer.py:81]:
```suggestion
        if not token_false and not token_true:
```

File: pygaggle/rerank/transformer.py
Comment (rodrigonogueira4) [inline pygaggle/rerank/transformer.py:88]: missing `f` in the f-string?

**Review by rodrigonogueira4 — APPROVED** [review]
Looks all good to me except for two minor comments. Also, could you please run pylint again (there are a few extra spaces between equal signs.

**Comment by rodrigonogueira4** [comment]
@ronakice, could you please take a look and once you are comfortable, merge it?

**Comment by rodrigonogueira4** [comment]
Hey @ronakice, I will merge this one but please let us know if you have any concerns

**Pull request merged**
